### PR TITLE
Bugfix #2170 main_v10.1 agg_ecnt

### LINKED ISSUE #2170: Fix Stat-Analysis to aggregate the ECNT ME and RMSE values correctly.
## Describe the Problem ##
This issue was raised via METplus discussions:
https://github.com/dtcenter/METplus/discussions/1632

When aggregating multiple ECNT lines together, Stat-Analysis reports ME = RMSE = 0 in the output even when it should not, based on the inputs.

### Expected Behavior ###
Stat-Analysis ```-job aggregate -line_type ECNT``` should correctly compute aggregated ME and RMSE scores rather than always reporting them as 0.

### Environment ###
Describe your runtime environment:
*1. Machine: Mac Laptop, but reproducible on any*
*2. OS: MacOS, but not OS-dependent*
*3. Software version number(s): Reproduced with MET versions 10.0, 10.1, and the latest in develop*

### To Reproduce ###
Describe the steps to reproduce the behavior:
*1. Download [sample_ecnt.txt](https://github.com/dtcenter/MET/files/8788151/sample_ecnt.txt)*
*2. Run:*
```
stat_analysis -lookin sample_ecnt.txt -job aggregate -line_type ECNT
```
*3. Observe ME = RMSE = 0 in the output*
```
COL_NAME: TOTAL N_ENS    CRPS CRPSS     IGN ME RMSE  SPREAD ME_OERR RMSE_OERR SPREAD_OERR SPREAD_PLUS_OERR CRPSCL CRPS_EMP CRPSCL_EMP CRPSS_EMP
    ECNT:     2     6 6.88929    NA 5.91505  0    0 7.34208      NA        NA          NA               NA     NA  3.47321         NA        NA
```

### Relevant Deadlines ###
*List relevant project deadlines here or state NONE.*

### Funding Source ###
*Define the source of funding and account keys here or state NONE.*

### Assignee ###
- [x] Select **engineer(s)** or **no engineer** required
- [x] Select **scientist(s)** or **no scientist** required: None required.

### Labels ###
- [x] Select **component(s)**
- [x] Select **priority**
- [x] Select **requestor(s)**

### Projects and Milestone ###
- [x] Select **Organization** level **Project** for support of the current coordinated release
- [x] Select **Repository** level **Project** for development toward the next official release or add **alert: NEED PROJECT ASSIGNMENT** label
- [x] Select **Milestone** as the next bugfix version

## Define Related Issue(s) ##
Consider the impact to the other METplus components.
- [x] [METplus](https://github.com/dtcenter/METplus/issues/new/choose), [MET](https://github.com/dtcenter/MET/issues/new/choose), [METdatadb](https://github.com/dtcenter/METdatadb/issues/new/choose), [METviewer](https://github.com/dtcenter/METviewer/issues/new/choose), [METexpress](https://github.com/dtcenter/METexpress/issues/new/choose), [METcalcpy](https://github.com/dtcenter/METcalcpy/issues/new/choose), [METplotpy](https://github.com/dtcenter/METplotpy/issues/new/choose)
No impacts.

## Bugfix Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [ ] Complete the issue definition above, including the **Time Estimate** and **Funding Source**.
- [x] Fork this repository or create a branch of **main_\<Version>**.
Branch name: `bugfix_<Issue Number>_main_<Version>_<Description>`
- [x] Fix the bug and test your changes.
- [x] Add/update log messages for easier debugging.
- [x] Add/update unit tests.
- [x] Add/update documentation.
- [x] Push local changes to GitHub.
- [x] Submit a pull request to merge into **main_\<Version>**.
Pull request: `bugfix <Issue Number> main_<Version> <Description>`
- [x] Define the pull request metadata, as permissions allow.
Select: **Reviewer(s)** and **Linked issues**
Select: **Organization** level software support **Project** for the current coordinated release
Select: **Milestone** as the next bugfix version
- [ ] Iterate until the reviewer(s) accept and merge your changes.
- [ ] Delete your fork or branch.
- [x] Complete the steps above to fix the bug on the **develop** branch.
Branch name:  `bugfix_<Issue Number>_develop_<Description>`
Pull request: `bugfix <Issue Number> develop <Description>`
Select: **Reviewer(s)** and **Linked issues**
Select: **Repository** level development cycle **Project** for the next official release
Select: **Milestone** as the next official version
- [ ] Close this issue.

### PR DESCRIPTION
**Pull request by JohnHalleyGotway**
## Expected Differences ##

- [x] Do these changes introduce new tools, command line arguments, or configuration file options? **[No]**</br>
If **yes**, please describe:</br>

- [x] Do these changes modify the structure of existing or add new output data types (e.g. statistic line types or NetCDF variables)? **[No]**</br>
If **yes**, please describe:</br>

## Pull Request Testing ##

- [x] Describe testing already performed for these changes:</br>
Manually tested to confirm that the aggregated ECNT ME and RMSE written by Stat-Analysis are no longer a constant value of 0. Note that this same issue affects the ME_OERR and RMSE_OERR columns, but those were already set to NA.

- [x] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>
1. Please inspect the code change.
2. Please confirm that the unit tests flag a difference in the output. Download the diff files and confirm see that the ME and RMSE output columns of the Stat-Analysis agg ecnt job are the only diffs.
3. Please review the doc updates about percentile thresholds. See them on RTD [here](https://met.readthedocs.io/en/bugfix_2170_main_v10.1_agg_ecnt/Users_Guide/config_options.html#configuration-file-overview).

- [x] Do these changes include sufficient documentation updates, ensuring that no errors or warnings exist in the build of the documentation? **[Yes]**

- [x] Do these changes include sufficient testing updates? **[Yes]**

- [x] Will this PR result in changes to the test suite? **[Yes]**</br>
If **yes**, describe the new output and/or changes to the existing output:</br>
It corrects the output of an existing Stat-Analysis test.

- [x] Please complete this pull request review by **[Fri 6/10/22]**.</br>

## Pull Request Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [x] Review the source issue metadata (required labels, projects, and milestone).
- [x] Complete the PR definition above.
- [x] Ensure the PR title matches the feature or bugfix branch name.
- [x] Define the PR metadata, as permissions allow.
Select: **Reviewer(s)**
Select: **Organization** level software support **Project** or **Repository** level development cycle **Project**
Select: **Milestone** as the version that will include these changes
- [x] After submitting the PR, select **Linked issues** with the original issue number.
- [x] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [x] Close the linked issue and delete your feature or bugfix branch from GitHub.

### CODE REVIEW

File: met/docs/Users_Guide/config_options.rst [hunk 1]
```diff
@@ -58,6 +58,13 @@ The configuration file language supports the following data types:
     and 10 and "==1||==2" defines numbers exactly equal to 1 or 2.
      
 * Percentile Thresholds:
+
+  * A threshold type (<, <=, ==, !-, >=, or >), followed by a percentile
+    type description (SFP, SOP, SCP, USP, CDP, or FBIAS), followed by a
+    numeric value, typically between 0 and 100.
+
+  * Note that the two letter threshold type abbreviations (lt, le, eq, ne,
+    gt, gt) are not supported for percentile thresholds.
```
Comment (jprestop) [inline met/docs/Users_Guide/config_options.rst]:
```suggestion
    ge, gt) are not supported for percentile thresholds.
```

**Comment by JohnHalleyGotway** [comment]
Reassigning to @jprestop. I didn't realize that @j-opatz was out of the office this week.

File: met/docs/Users_Guide/config_options.rst [hunk 2]
```diff
@@ -58,6 +58,13 @@ The configuration file language supports the following data types:
     and 10 and "==1||==2" defines numbers exactly equal to 1 or 2.
      
 * Percentile Thresholds:
+
+  * A threshold type (<, <=, ==, !-, >=, or >), followed by a percentile
```
Comment (jprestop) [inline met/docs/Users_Guide/config_options.rst]:
```suggestion
  * A threshold type (<, <=, ==, !=, >=, or >), followed by a percentile
```

**Review by jprestop — APPROVED** [review]
I inspected the code change, confirmed that the unit tests flagged a difference in the output, only in the ME and RMSE output columns of the Stat-Analysis agg ecnt job, and reviewed the doc updates about percentile thresholds. I made a couple of suggested corrections for "gt" to "ge" and "!-" to "!=", and I think the "!=" still needs to be made before merging.  I approve this request.

**Comment by JohnHalleyGotway** [comment]
Thanks Julie. No idea why I made those silly typos. Thanks for catching them! Guess I was trying to multi task too much.

**Pull request merged**